ksqldb-engine/src/test/java/io/confluent/ksql/serde/SerdeFeaturesFactoryTest.java

```diff
@@ -390,10 +411,55 @@ public void shouldNotConvertFormatForMulticolKeysWhenSanitizingIfDisallowed() {
         SerdeFeatures.of());
 
     // When:
-    final KeyFormat sanitized = SerdeFeaturesFactory.sanitizeKeyFormat(format, 2, false);
+    final KeyFormat sanitized = SerdeFeaturesFactory.sanitizeKeyFormat(format, MULTI_SQL_TYPES, false);
 
     // Then:
     assertThat(sanitized.getFormatInfo(), equalTo(FormatInfo.of(KafkaFormat.NAME)));
     assertThat(sanitized.getFeatures(), equalTo(SerdeFeatures.of()));
   }
+
+  @Test
+  public void shouldConvertKafkaFormatForSingleKeyWithNonPrimitiveType() {
```

Comment:
What about `shouldConvertKafkaFormatForSingleKeyWithPrimitiveType`, is that covered elsewhere? Ditto beow.